pkg/scm/gitlab/client_actioner.go

```diff
@@ -153,6 +155,81 @@ func (c *Client) ApplyStep(ctx context.Context, evalContext scm.EvalContext, upd
 
 		return err
 
+	case "assign_reviewers":
+		source, err := step.RequiredStringEnum("source", "codeowners")
+		if err != nil {
+			return err
+		}
+
+		desiredLimit, err := step.RequiredInt("limit")
+		if err != nil {
+			return err
+		}
+
+		mode, err := step.OptionalStringEnum("mode", "linear", "linear", "random")
```

Comment:
using the pattern used for Strings for validating other types in the actions schema. maybe should start breaking up actions in their own class per scm platform

Reply:
thinking the schema itself should be generic, but how the actions are ran should be left up to the scm platform action instance, avoids wrapping most the client and context to make it generic enough. then its up to the action instance per platform to let the actioner know if its supported or not for the platform

Reply:
Yeah, these also need to be encoded into the JSONSchema used for the editor.

`func (step ActionStep) JSONSchema() *jsonschema.Schema {` already exists as an example of how to encode custom requirements into the schema validation - the underlying library (https://github.com/invopop/jsonschema) got some okay-ish docs on how to configure some of this via JSON tags on struct fields - let me know if its causing a bunch of pain and I can look into it - it was painful for me the first time around (and probably still is!)